packages/dynamite/dynamite/lib/src/helpers/version_checker.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:build/build.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+final dependencies = {
+  'built_collection': Version.parse('5.0.0'),
+  'built_value': Version.parse('8.0.0'),
+  'collection': Version.parse('1.0.0'),
+  'dynamite_runtime': Version.parse('0.0.1'),
+  'meta': Version.parse('1.0.0'),
+  'universal_io': Version.parse('2.0.0'),
+  'uri': Version.parse('1.0.0'),
+};
+
+final devDependencies = {
+  'built_value_generator': Version.parse('8.8.0'),
+};
+
+/// Checks whether the correct version of the dependencies are present in the pubspec.yaml file.
+Future<({bool hasFatal, String messages})> helperVersionCheck(final BuildStep buildStep) async {
+  final pubspecAsset = AssetId(buildStep.inputId.package, 'pubspec.yaml');
+
+  if (!await buildStep.canRead(pubspecAsset)) {
+    throw Exception(
+      'Failed to read the pubspec.yaml file. '
+      'Version constraints of required packages can not be validated.',
```

Comment:
Can you make these just one string? I think this is quite ugly (especially the trailing space)